azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -681,18 +671,28 @@ public static void main(String[] args) throws Exception {
     StdOutErrRedirect.redirectOutAndErrToLog();
 
     logger.info("Starting Jetty Azkaban Web Server...");
-    Props azkabanSettings = AzkabanServer.loadProps(args);
+    Props props = AzkabanServer.loadProps(args);
 
-    if (azkabanSettings == null) {
+    if (props == null) {
       logger.error("Azkaban Properties not loaded. Exiting..");
       System.exit(1);
     }
 
-    final Server server = createServer(azkabanSettings);
+    /* Initialize Guice Injector */
+    SERVICE_PROVIDER.setInjector(Guice.createInjector(
+        new AzkabanCommonModule(props),
+        new AzkabanWebServerModule()
+    ));
 
-    app = new AzkabanWebServer(server, azkabanSettings);
+    launch(props);
+  }
+
+  public static void launch(Props azkabanSettings) throws Exception {
+    /* This creates the Web Server instance */
+    app = SERVICE_PROVIDER.getInstance(AzkabanWebServer.class);
 
-    prepareAndStartServer(azkabanSettings, server);
+    // TODO refactor code into ServerProvider
```

Comment:
In the future, please add your name to the todo comment. See our todo code convention.